### PR TITLE
Add batches endpoints

### LINKED ISSUE #695: [v1.12.0] New batch routes
Related to https://github.com/meilisearch/integration-guides/issues/307

### Explanation of the feature

- New `GET /batches` & `GET /batches/:uid` routes
- New field in `task` object: `batchUid`

Usage: https://meilisearch.notion.site/Enhance-visibility-on-batched-tasks-1194b06b651f810b8fe0fab5d72846a8?pvs=74

### TODO

👉 Check carefully the usage above to ensure your new methods and objects integrate all the functionnalities

- [ ] Create two new methods `getBatch()` and `getBatches()` triggering the appropriate new route (check out `getTask` and `getTasks` since the behavior will be pretty similar)
- [ ] A new `batch` object will probably be needed to be created.
- [ ] Ensure the `task` object can accept the `batchUid` field
- [ ] Add tests

⚠️ Make PRs pointing to `bump-meilisearch-v1.12.0` and NOT `main`. Please do 1 PR for all of these changes, and not several.

### PR DESCRIPTION
**Pull request by Strift**
# Pull Request

## Related issue
Fixes #695

### CODE REVIEW

File: src/Contracts/BatchesResults.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    private int $from;
+
+    /**
+     * @var non-negative-int
+     */
+    private int $total;
+
+    public function __construct(array $params)
+    {
+        parent::__construct($params['results'] ?? []);
```
Comment (norkunas) [inline src/Contracts/BatchesResults.php:31]: Are there any cases when results will not be set?

File: src/Endpoints/Delegates/HandlesBatches.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    protected Batches $batches;
+
+    public function getBatch($uid): array
+    {
+        return $this->batches->get($uid);
+    }
+
+    public function getBatches(?BatchesQuery $options = null): BatchesResults
+    {
+        $query = isset($options) ? $options->toArray() : [];
```
Comment (norkunas) [inline src/Endpoints/Delegates/HandlesBatches.php:22]:
```suggestion
        $query = null !== $options ? $options->toArray() : [];
```

File: src/Contracts/BatchesResults.php (hunk 1 above)
Comment (Strift) [reply src/Contracts/BatchesResults.php:31]: As far as I understand, the API should always return a `results` array. 

But I saw in the `TasksResults` class that there was a check for the field being missing. So I used the same logic.

Comment (norkunas) [reply src/Contracts/BatchesResults.php:31]: So it's a dead code then :slightly_smiling_face:

Comment (Strift) [reply src/Contracts/BatchesResults.php]: I updated this line ✅

**Review by norkunas — APPROVED** [review]

**Pull request merged**